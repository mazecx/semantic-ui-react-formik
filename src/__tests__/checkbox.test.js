import React from 'react'
import {render, cleanup} from 'react-testing-library'
import Form, {Checkbox} from '../'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup)

const FormExample = props => (
  <Form initialValues={{field: []}} {...props}>
    {() => (
      <React.Fragment>
        <Checkbox id="field-1" label="My value 1" name="field" value="1" />
        <Checkbox id="field-2" label="My value 2" name="field" value="2" />
      </React.Fragment>
    )}
  </Form>
)

test('should render all checkbox', () => {
  const {getByText, container} = render(<FormExample />)
  expect(getByText(/my value 1/i))
  expect(getByText(/my value 2/i))
  expect(container.firstChild).toMatchSnapshot()
})