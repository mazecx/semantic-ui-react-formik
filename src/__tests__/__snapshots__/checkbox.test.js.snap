// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all checkbox 1`] = `
<form
  class="ui form"
>
  <div
    class="field"
  >
    <div
      class="ui checkbox"
    >
      <input
        class="hidden"
        id="field-1"
        readonly=""
        tabindex="0"
        type="checkbox"
        value=""
      />
      <label
        for="field-1"
      >
        My value 1
      </label>
    </div>
  </div>
  <div
    class="field"
  >
    <div
      class="ui checkbox"
    >
      <input
        class="hidden"
        id="field-2"
        readonly=""
        tabindex="0"
        type="checkbox"
        value=""
      />
      <label
        for="field-2"
      >
        My value 2
      </label>
    </div>
  </div>
</form>
`;
